Add documentation about parent attribute (#703)
psss/tmt: docs/contribute.rst

    │   └── GuestTestcloud
    └── Run

Attributes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Object hierarchy is following: Run -> Plans -> Steps -> Plugin, where the Run is

Reviewer: Please, use double ``Quotes`` for for all class names.

on the top of this hierarchy. The objects have the ``parent`` attribute, that is

> pointing on a parent from which the current object inherits from.

Reviewer: `Plan` does not inherit from `Run`, they are included. Imagine `Run` as a container of plans. Let's use "contain" here to prevent possible confusion.

> ``Node`` is a class which inherits from ``Common`` and points at the original fmf tree’s data.

Reviewer: The class inheritance is covered above. Let's mention the `node` attribute instead, which contains reference to the original fmf node.

> Class ``tmt.Tree`` is initialized with the ``fmf.Tree`` and saves the original fmf tree and it’s raw metadata.

Reviewer: ```suggestion
The original ``fmf.Tree`` from which the ``tmt.Tree`` was initialized is available under the ``tree`` property.
```

    └── Run

Attributes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Object hierarchy is following: ``Run`` -> ``Plans`` -> ``Steps``
-> ``Plugins``, where the ``Run`` is on the top of this hierarchy.
The objects have the ``parent`` attribute, that is pointing to
the parent in which current class is contained.

The ``node`` attribute of ``Test``, ``Plan`` and ``Story``

> instances references the original fmf metadata node from which

Reviewer 2: not sure if I understand this. `instances references the original fmf metadata node` - from my perspective my understanding is that node contains same data as `Tree.tree`.
Maybe some different sentence like: `instances copies FMF data from original metadata node` 

if I understand it well @psss . :-)

Reviewer: The `node` references the corresponding leaf of the fmf metadata tree from which test/plan/story has been created. It does not copy any information. Maybe using `leaf` would make it a bit more clear?

Reviewer: What about this:

> The `node` attribute ... references the original leaf node of the fmf metadata tree from which the respective test/plan/story has been created.

Does this sound more clear?

Reviewer 2: yes, it is more clear now.

Reviewer: Thanks, updated.
